packages/Search/test/tstBoostGeometryAdapters.cpp

```diff
@@ -0,0 +1,249 @@
+    // constructed "empty" box is not valid, it will still calculate its
+    // centroid.  Admittedly, the result (centroid at origin) is garbage :)
+    dtk::Box empty_box = {};
+    TEST_ASSERT( !bg::is_valid( empty_box ) );
+    TEST_ASSERT( equals( bg::return_centroid<dtk::Point>( empty_box ),
+                         {{0., 0., 0.}} ) );
+    TEST_ASSERT( equals( dtkReturnCentroid( empty_box ), {{0., 0., 0.}} ) );
+
+    dtk::Box unit_box = {{{0., 0., 0.}}, {{1., 1., 1.}}};
+    TEST_ASSERT(
+        equals( bg::return_centroid<dtk::Point>( unit_box ), {{.5, .5, .5}} ) );
+    TEST_ASSERT( equals( dtkReturnCentroid( unit_box ), {{.5, .5, .5}} ) );
+
+    // NOTE DTK does not have an overload of centroid() for Point at the
+    // moment.
+    dtk::Point a_point = {{1., 2., 3.}};
+    TEST_ASSERT(
+        equals( bg::return_centroid<dtk::Point>( a_point ), a_point ) );
+    // TEST_ASSERT( equals( dtk::centroid(
+    //     []( dtk::Point p ) {
+    //         dtk::Point c;
+    //         dtk::centroid( p, c );
+    //         return c;
+    //     }( a_point ),
+    //     a_point ) ) );
+}
+
+TEUCHOS_UNIT_TEST( BoostGeometryAdapters, is_valid )
+{
+    // NOTE "empty" box is considered as valid in DataTransferKit but it is
+    // not according to Boost.Geometry
+    dtk::Box empty_box = {};
+    TEST_ASSERT( dtk::isValid( empty_box ) );
+    std::string message;
+    TEST_ASSERT( !bg::is_valid( empty_box, message ) );
+    TEST_EQUALITY( message, "Box has corners in wrong order" );
+
+    // Same issue with infinitesimal box around a point (here the origin)
+    dtk::Box a_box = {{{0., 0., 0.}}, {{0., 0., 0.}}};
+    TEST_ASSERT( dtk::isValid( a_box ) );
+    TEST_ASSERT( !bg::is_valid( a_box, message ) );
+    TEST_EQUALITY( message, "Geometry has wrong topological dimension" );
+
+    dtk::Box unit_box = {{{0., 0., 0.}}, {{1., 1., 1.}}};
+    TEST_ASSERT( dtk::isValid( unit_box ) );
+    TEST_ASSERT( bg::is_valid( unit_box ) );
+
+    dtk::Box invalid_box = {{{1., 2., 3.}}, {{4., NAN, 6.}}};
+    TEST_ASSERT( !dtk::isValid( invalid_box ) );
+    TEST_ASSERT( !bg::is_valid( invalid_box, message ) );
+    TEST_EQUALITY( message,
+                   "Geometry has point(s) with invalid coordinate(s)" );
+
+    dtk::Point a_point = {{1., 2., 3.}};
+    TEST_ASSERT( dtk::isValid( a_point ) );
+    TEST_ASSERT( bg::is_valid( a_point ) );
+
+    auto const infty = std::numeric_limits<double>::infinity();
+    dtk::Point invalid_point = {{infty, 1.41, 3.14}};
+    TEST_ASSERT( !dtk::isValid( invalid_point ) );
+    TEST_ASSERT( !bg::is_valid( invalid_point, message ) );
+    TEST_EQUALITY( message,
+                   "Geometry has point(s) with invalid coordinate(s)" );
+
+    // Also Boost.Geometry has a is_empty() algorithm but it has a different
+    // meaning, it checks whether a geometry is an empty set and always
+    // return
+    // false for a point or a box.
```

Comment:
can you move the comment on the same line as `return`

Reply:
Will do.  Auto indent didn't do its job very well :)